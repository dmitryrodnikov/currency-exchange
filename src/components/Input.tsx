import * as React from 'react';
import {ChangeEvent} from 'react';
import styled from 'styled-components';
import {COLORS, FONT_SIZE} from '../constants/Theme';

interface InputProps {
    value: string;
    onChange?: InputChangeHandler;
}

export type InputChangeHandler = (value: number) => void;

const StyledInput = styled.input`
    border: none;
    border-radius: 8px;
    padding: 0 8px;
    box-sizing: border-box;
    text-align: right;
    outline: none;
    height: 36px;
    width: 100%;
    color: ${COLORS.black};
    font-size: ${FONT_SIZE.l};
    font-weight: bold;
    transition: 300ms background-color;
    
    /* Remove input controls in Webkit browsers */
    &::-webkit-inner-spin-button,
    &::-webkit-outer-spin-button {
        -webkit-appearance: none;
        margin: 0;
    }
    /* Remove input controls in Firefox */
    -moz-appearance:textfield;
    
    &:hover,
    &:focus {
        background-color: ${COLORS.green}10;
    }
`;

export class Input extends React.PureComponent<InputProps> {
    public render() {
        return (
            <StyledInput
                type='number'
                value={this.props.value}
                onChange={this.onChangeHandler}
            />
        );
    }

    private onChangeHandler = (e: ChangeEvent<HTMLInputElement>): void => {
        const {onChange} = this.props;
        if (onChange) {
            onChange(Number(e.target.value));
        }
    }
}
