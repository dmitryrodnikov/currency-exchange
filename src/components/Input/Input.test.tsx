import React from 'react';
import {shallow, render} from 'enzyme';
import {Input} from './Input';

const INPUT_VALUE= 'Test';

describe('Input', () => {
    it('Shallow render snapshots are equal', () => {
        const wrapper = shallow(
            <Input value={INPUT_VALUE}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('Render snapshots are equal', () => {
        const wrapper = render(
            <Input value={INPUT_VALUE}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
