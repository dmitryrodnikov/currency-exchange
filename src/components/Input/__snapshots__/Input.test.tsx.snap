// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Render snapshots are equal 1`] = `
<input
  class="sc-bdVaJa jSDdor"
  type="text"
  value=""
/>
`;

exports[`Input Shallow render snapshots are equal 1`] = `
<Styled(NumberFormat)
  allowNegative={false}
  decimalScale={2}
  onChange={[Function]}
  value="Test"
/>
`;
